include/cutlass/fast_math.h

```diff
@@ -767,6 +776,61 @@ half_t fast_tanh(half_t x) {
 
 /////////////////////////////////////////////////////////////////////////////////////////////////
 
+template <typename T>
+struct fast_exp_op {
+  CUTLASS_HOST_DEVICE
+  T operator()(T const &rhs) const {
+    return fast_exp(rhs);
+  }
+};
+
+#if defined(__CUDA_ARCH__) && (__CUDACC_VER_MAJOR__ >= 10) && (__CUDA_ARCH__ >= 750)
+template <int N>
+struct fast_exp_op<Array<half_t, N>> {
+  CUTLASS_DEVICE
+  Array<half_t, N> operator()(Array<half_t, N> const &rhs) const {
+
+    Array<half_t, N> result;
+
+    // use x2 specialization
+    __half2 const *in  = reinterpret_cast<__half2 const *>(&rhs);
+    __half2 *out = reinterpret_cast<__half2 *>(&result);
+
+    CUTLASS_PRAGMA_UNROLL
+    for (int i = 0; i < N / 2; ++i) {
+      out[i] = ::h2exp(in[i]);
+    }
+
+    // residual
+    if (N % 2) {
+      half_t last = rhs[N - 1];
+      result[N - 1] = half_t(::hexp(last.raw()));
```

Comment:
my bad.

these line can be replaced by

```
      result[N - 1] = ::hexp(rhs[N - 1].to_half());
```

Reply:
That still fails with
```
cutlass/include/cutlass/fast_math.h(807): error: class "cutlass::Array<T, N, false>::const_reference [with T=cutlass::half_t, N=8]" has no member "to
_half"                                                                                                       
```
You also missed `half_t` before the assignment.

Reply:
hmm. `raw()` returns a `uint16_t` not a `fp16` which is not what we want. Can you try 
`result[N - 1] = half_t(::hexp(last.to_half()));`?